File: String/string_easy.c
#include <string.h>
#include <stdio.h>
#include <stdlib.h>

////////////////////////////////////////REVERSE STRING////////////////////////////////////////////////
char* reverseString(char* s) {
	int length = 0;
	while (s[length] != '\0') {
		length++;
	}

	char* result =  calloc(length+1, sizeof(char));
	for (int i = 0; i < length; i++){
		result[i]= s[length-1-i];
	}
	result[length] = '\0'; //important!!!
	return result;
}

////////////////////////////////////////REVERSE STRING////////////////////////////////////////////////

File: String/string_easy.py
#############################################reverse string###################################
def reverseString(s):
	result = ""
	length = len(s)
	i = length-1

	while (i >= 0):
		result += s[i]
		i -= 1
	return result
#############################################reverse string###################################

#############################################reverse integer###################################
def reverse(x):
	minNum = -1 << 31;
    maxNum = (1 << 31) - 1;
    
    sign = (-1) if (x < 0) else 1;

    result = ''

    digit_list = list(str(abs(x)))
    

    for digit in list(reversed(digit_list)):
        result += digit

    resultInt = sign*int(result);
    
    if ((resultInt < minNum) or (resultInt > maxNum)): return 0;
    return resultInt;

#############################################reverse integer###################################